src/service/users.js

```diff
@@ -237,3 +237,20 @@ export const checkFollowStatusBatch = async (targetUserIds = [], axiosOptions =
     );
   }
 };
+
+export const checkFollowStatus = async (channelId) => {
+  if (!channelId) return Promise.reject(new Error('Target User ID is empty'));
+
+  try {
+    const result = await api.get(`/users/check-follow-by-channelId?channelId=${channelId}`);
+    if (result?.status === 200) {
+      return Promise.resolve(result?.data?.data);
+    }
+
+    return Promise.reject(result?.data?.message);
+  } catch (e) {
+    return Promise.reject(
+      e?.response?.data?.message || "Can't check follow status, please try again later"
+    );
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `checkFollowStatus` function seems to be well implemented. It checks if the `channelId` is provided, makes an API call, and handles both success and error responses appropriately. However, it's recommended to encode the `channelId` in the URL to prevent potential security issues related to URL injection.

```diff
- const result = await api.get(`/users/check-follow-by-channelId?channelId=${channelId}`);
+ const encodedChannelId = encodeURIComponent(channelId);
+ const result = await api.get(`/users/check-follow-by-channelId?channelId=${encodedChannelId}`);
```
This change ensures that the `channelId` is properly encoded before being included in the URL, which can help prevent potential security vulnerabilities.


<!-- This is an auto-generated comment by OSS CodeRabbit -->